### ISSUE TITLE
Git hooks should only check staged/committed files

### ISSUE THREAD

**Issue by programLyrique** [PRL-PRG/r-compile-server#10]
The git hooks perform checks and tests on changes they are not supposed to check.

The pre-commit one should only perform checks (such as the formatting one) on the files added to the staging area.

The pre-push should perform checks and tests on the last commit, not on changes that have not been committed.

**Comment by Jakobeha** [comment]
The pre-push hook was removed, but Git doesn't propagate hooks so you must run `rm .git/hooks/pre-push.sh`. Sorry about that.

I created a PR (https://github.com/PRL-PRG/r-compile-server/pull/11) to fix the pre-commit so it only applies to staged files unless everything is staged, and can also auto-format the staged files instead of just throwing an error. I kept it so that it formats everything if everything is staged (even non-committed files) because it's apparently recommended, since spotless already caches (see https://github.com/diffplug/spotless/issues/178).
